src/legacy/core_plugins/kibana/migrations/migrations.test.js

```diff
@@ -1180,6 +1180,121 @@ Array [
       expect(migratedDoc).toEqual({ attributes: { visState: JSON.stringify(expected) } });
     });
   });
+  describe('7.5.0 tsvb split_filters migration', () => {
+    const migrate = doc => migrations.visualization['7.5.0'](doc);
+    const generateDoc = ({ params }) => ({
+      attributes: {
+        title: 'My Vis',
+        description: 'This is my super cool vis.',
+        visState: JSON.stringify({ params }),
+        uiStateJSON: '{}',
+        version: 1,
+        kibanaSavedObjectMeta: {
+          searchSourceJSON: '{}',
+        },
+      },
+    });
+    it('should change series item filters from a string into an object for all filters', () => {
+      const params = {
+        type: 'timeseries',
+        filter: {
+          query: 'bytes:>1000',
+          language: 'lucene'
+        },
+        series: [
+          {
+            split_filters: [{ filter: 'bytes:>1000' }],
+          },
+        ]
+      };
+      const timeSeriesDoc = generateDoc({ params: params });
+      const migratedtimeSeriesDoc = migrate(timeSeriesDoc);
+      const timeSeriesParams = JSON.parse(migratedtimeSeriesDoc.attributes.visState).params;
+      expect(Object.keys(timeSeriesParams.filter)).toEqual(
+        expect.arrayContaining(['query', 'language'])
+      );
```

Comment:
Nitpick: could we perhaps `toEqual` the full object here instead of just the keys? Like `toEqual({ query: 'bytes:>1000', language: 'lucene' })`? I think it would just make for a bit safer test. (The same applies for the test below.)

Reply:
[done](https://github.com/elastic/kibana/pull/49000/files#diff-5afd24840636884e72ed2920f5e67a71R1217)